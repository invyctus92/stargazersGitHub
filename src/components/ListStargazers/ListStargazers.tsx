/**
 * FlatList with the list of stargazers
 *
 * @format
 */

import React, { type PropsWithChildren } from "react";
import { FlatList, Image, StyleSheet, Text, View } from "react-native";

// a Stargazer
interface Stargazer {
    avatar_url: string;
    login: string;
    id: string;
}

const SingleElementStargazer: React.FC<
    PropsWithChildren<{
        item: Stargazer;
    }>
> = ({ item }) => {
    return (
        <View
            style={styles.viewStargazer}
        >
            <Image
                source={{ uri: item.avatar_url }}
                style={styles.imageStargazer}
            />
            <Text>{item.login}</Text>
        </View>
    );
};

const NoData: React.FC<PropsWithChildren<{}>> = () => {
    return (
        <View
            style={styles.styleNoData}
        >
            <Text>Here you will display the Stargazers found</Text>
        </View>
    );
};

const ListStargazers: React.FC<
    PropsWithChildren<{
        stargazers: Stargazer[];
    }>
> = ({ stargazers = [] }) => {
    return (
        <FlatList
            style={styles.styleList}
            removeClippedSubviews={false}
            showsVerticalScrollIndicator={false}
            ListEmptyComponent={<NoData />}
            data={stargazers}
            keyExtractor={(elem: Stargazer) => elem.id}
            initialNumToRender={7}
            renderItem={({ item, index }) => {
                return <SingleElementStargazer item={item} />;
            }}
        />
    );
};

const styles = StyleSheet.create({
    styleList: {
        backgroundColor: "white",
        width: "100%",
        height: "100%",
    },
    styleNoData: {
        flex: 1,
        flexDirection: "column",
        alignItems: "center",
        height: 200,
        alignContent: "center",
        justifyContent: "center",
    },
    imageStargazer: {
        margin: 10,
        width: 100,
        height: 100,
        borderRadius: 50,
    },
    viewStargazer: {
        flex: 1,
        flexDirection: "row",
        alignItems: "center",
    }
});

export default ListStargazers;
