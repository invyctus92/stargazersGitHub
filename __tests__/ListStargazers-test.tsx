

import React from 'react';
import renderer from 'react-test-renderer';
import ListStargazers from '../src/components/ListStargazers/ListStargazers';

test('renders correctly', () => {
  const tree = renderer.create(<ListStargazers stargazers={[]} />).toJSON();
  expect(tree).toMatchSnapshot();
});

