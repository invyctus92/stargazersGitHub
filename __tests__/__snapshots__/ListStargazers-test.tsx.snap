// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  ListEmptyComponent={<NoData />}
  data={Array []}
  getItem={[Function]}
  getItemCount={[Function]}
  initialNumToRender={7}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={50}
  showsVerticalScrollIndicator={false}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "white",
      "height": "100%",
      "width": "100%",
    }
  }
  viewabilityConfigCallbackPairs={Array []}
>
  <View>
    <View
      style={
        Object {
          "alignContent": "center",
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "height": 200,
          "justifyContent": "center",
        }
      }
    >
      <Text>
        Here you will display the Stargazers found
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
